### PR TITLE
Fix race condition when binding to PORT_ANY

### LINKED ISSUE #202: Race condition in _get_available_port
*PyBluez is not under active development but we are seeking new contributors
to investigate bugs and submit patches.*

## System

 - **Operating System:** Linux
 - **Hardware:** external Bluetooth adaptor
 - **PyBluez Version:** latest (0.18)

## Issue

Please provide information necessary to reproduce the issue, including sample
code.

If you have long logs, please post them on https://gist.github.com and link to
the Gist here.

The issue is version-independent, and it is as follows:

1. An instance of python tries to bind to a specified address with PORT_ANY.
2. The function `_get_available_port` returns some unused port for it.
3. Someone (possibly a different python instance, or any other program) binds successfully to the port we wanted to bind to.
4. Then we try to bind to it, and execution fails.

A simple example:
```py
import bluetooth, os, subprocess

src_addr = subprocess.check_output(["hcitool", "dev"]).split()[-1]
os.fork()

sock = bluetooth.BluetoothSocket(bluetooth.L2CAP)
sock.bind((src_addr, 0))
```

Expected exception in at least one instance:
```py
Traceback (most recent call last):
  File "test.py", line 7, in <module>
    sock.bind((src_addr, 0))
  File "/usr/lib64/python2.7/site-packages/bluetooth/bluez.py", line 140, in bind
    return self._sock.bind (addrport)
_bluetooth.error: (98, 'Address already in use')
```
Same with `proto=bluetooth.RFCOMM`

### PR DESCRIPTION
**Pull request by Arusekk**
Closes #202

### CODE REVIEW

File: bluetooth/bluez.py [hunk 1]
```diff
@@ -203,13 +186,20 @@ def accept (self):
     accept.__doc__ = _bt.btsocket.accept.__doc__
 
     def bind (self, addrport):
```
Comment (karulis) [inline bluetooth/bluez.py]: I am not sure if it is nice API design that user asks for particula port number and behind the scenes gest some different number?

Comment (karulis) [inline bluetooth/bluez.py:188]: Sorry for comment deletion Arusek, I have noticed it is port 0 only after I have added comment.
Looks good, I wonder how it works on Windows, do we also have race condition there

**Comment by Arusekk** [comment]
I may need to rethink the design, as not all protocols give a two-element tuple, for example SCO.

**Comment by rgov** [comment]
Thanks for this contribution! Please let me know when it is ready to merge.

**Comment by Arusekk** [comment]
It should be ready now

**Comment by rgov** [comment]
The call to `range()` is going to use a lot of memory on Python 2, where maybe it should be `xrange()`, but apparently there's no good way to do this in a concise way that's compatible with Python 2 and 3. At least it's a short-lived allocation, even if it's wasteful.

**Pull request merged**